tests/unit-tests/woocommerce-cli.php

```diff
@@ -0,0 +1,196 @@
+<?php
+/**
+ * Tests CLI scripts for WooComerce.
+ *
+ * @package Newspack\Tests
+ */
+
+use Newspack\WooCommerce_Cli;
+
+/**
+ * Tests the Webhooks functionality.
+ */
+class Newspack_Test_WooCommerce_Cli extends WP_UnitTestCase {
+	/**
+	 * A user to assign test subscriptions/orders.
+	 *
+	 * @var int
+	 */
+	private static $user_id = null;
+
+	/**
+	 * Set up before class.
+	 */
+	public static function set_up_before_class() { // phpcs:ignore Squiz.Commenting.FunctionComment.Missing
+		self::$user_id = wp_insert_user(
+			[
+				'user_login' => 'test_user',
+				'user_email' => 'test@example.com',
+				'user_pass'  => 'password',
+				'meta_input' => [
+					'first_name'     => 'John',
+					'last_name'      => 'Doe',
+					'wc_total_spent' => 100,
+				],
+			]
+		);
+	}
+
+	/**
+	 * Test a healthy subscription that has a future next_payment date.
+	 */
+	public function test_healthy_subscription() {
+		$subscription = new WC_Subscription(
+			[
+				'id'               => 1,
+				'status'           => 'active',
+				'billing_period'   => 'month',
+				'billing_interval' => 1,
+				'total'            => 10, // Amount of recurring payment.
+				'dates'            => [
+					'start'        => gmdate( 'Y-m-d H:i:s', strtotime( '-6 month' ) ),
+					'next_payment' => gmdate( 'Y-m-d H:i:s', strtotime( '+10 day' ) ),
+				],
+				'orders'           => [
+					wc_create_order(
+						[
+							'customer_id'    => self::$user_id,
+							'status'         => 'completed',
+							'total'          => 10,
+							'date_completed' => gmdate( 'Y-m-d H:i:s', strtotime( '-1 month' ) ),
+						]
+					),
+				],
+			]
+		);
+		$result = WooCommerce_Cli::calculate_next_payment_date( $subscription );
+
+		// Subscription wasn't processed.
+		$this->assertFalse( $result );
```

Comment:
Nit: instead of a code comment, the description can be an argument in the assertion. This will make test failures more readable. 

```suggestion
		$this->assertFalse( $result, Subscription wasn't processed. );
```

Reply:
Ah, good call—2603e8a updates the test assertions.